File: Python/Indamon.py
class Pet:
    # Constructor
    def __init__(self, name, hp, attack, defence, fainted):
        self._self = self
        self._name = name
        self._hp = hp
        self._attack = attack
        self._defence = defence
        self._fainted = fainted

    # No getters and setters since it seemes to be the python way.
     

    # Print info

    def printInfo(Pet):
        print("INFO")
        print("Name: ", Pet._name)
        print("Hp: ", Pet._hp)
        print("Attack: ", Pet._attack)
        print("Defence: ", Pet._defence)
        print("Fainted: ", Pet._fainted)

    def attack(self, defender):
        damage = self._attack/defender._defence
        print(damage, "damage has been dealt")
        print(defender._name, "have", (defender._hp - damage), "hp left")

File: Javascript/Indamon.mjs
class Indamon {
    #name;
    #hp;
    #attack;
    #defense;
    #fainted;

    get name() {
        return this.#name;}
    set name(newName) {
        this.#name = newName;}

    get hp() {
        return this.#hp;}
    set hp(newHp) {
        this.#hp = newHp;}
    
    get attack() {
        return this.#attack;}
    set attack(newAttack) {
        this.#attack = newAttack}

    get defense() {
        return this.#defense;}
    set defense(newDfense) {
        this.#defense = newDfense;}
    
    get fainted() {
        return this.#fainted;}
    set fainted(newFainted) {
        this.#fainted = newFainted;}

        
    constructor(name, hp, attack, defense, fainted) {
        this.name = name;
        this.hp = hp;
        this. attack = attack;
        this.defense = defense;
        this.fainted = fainted;
    }

    printInfo() {
        console.log("INFO")
        console.log("Indamon: ", this.name);
        console.log("Hp: ", this.hp);
        console.log("Attacl: ", this.attack);
        console.log("Defense: ", this.defense);
        console.log("Fainted: ", this.fainted);
    }
}

export default Indamon;